src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/compute/v2/AwsV2ComputePlugin.java

```diff
@@ -1,47 +1,459 @@
+		}
+	}
+
+	@Override
+	public boolean isReady(String instanceState) {
+		return AwsV2StateMapper.map(ResourceType.COMPUTE, instanceState).equals(InstanceState.READY);
+	}
+
+	@Override
+	public boolean hasFailed(String instanceState) {
+		return false;
+	}
+
+	protected ComputeInstance mountComputeInstance(Instance instance, List<Volume> volumes)
+			throws InstanceNotFoundException {
+
+		String id = instance.instanceId();
+		String cloudState = instance.state().nameAsString();
+		String name = instance.tags().get(FIRST_POSITION).value();
+		int cpu = instance.cpuOptions().coreCount();
+		int memory = getMemoryValueFrom(instance.instanceType());
+		int disk = getAllDisksSize(volumes);
+		List<String> ipAddresses = getIpAddresses(instance);
+		return new ComputeInstance(id, cloudState, name, cpu, memory, disk, ipAddresses);
+	}
+
+	protected List<String> getIpAddresses(Instance instance) {
+		List<String> ipAddresses = new ArrayList<String>();
+		for (int i = 0; i < instance.networkInterfaces().size(); i++) {
+			ipAddresses.add(instance.networkInterfaces().get(i).privateIpAddress());
+		}
+		return ipAddresses;
+	}
+
+	protected int getAllDisksSize(List<Volume> volumes) {
+		int size = 0;
+		for (Volume volume : volumes) {
+			size += volume.size();
+		}
+		return size;
+	}
+
+	protected int getMemoryValueFrom(InstanceType instanceType) {
+		int memory = 0;
+		for (AwsHardwareRequirements flavors : getFlavors()) {
+			if (flavors.getName().equals(instanceType.toString())) {
+				memory = flavors.getMemory();
+			}
+		}
+		return memory;
+	}
+
+	protected List<Volume> getInstanceVolumes(Instance instance, Ec2Client client) {
+		List<Volume> volumes = null;
+		DescribeVolumesResponse response;
+		List<String> volumeIds = getVolumeIds(instance);
+		for (String volumeId : volumeIds) {
+			response = getDescribeVolume(volumeId, client);
+			volumes = response.volumes();
+
+		}
+		return volumes;
+	}
+
+	protected DescribeVolumesResponse getDescribeVolume(String volumeId, Ec2Client client) {
+		DescribeVolumesRequest request = DescribeVolumesRequest.builder()
+				.volumeIds(volumeId)
+				.build();
+
+		return client.describeVolumes(request);
+	}
+
+	protected List<String> getVolumeIds(Instance instance) {
+		List<String> volumeIds = new ArrayList<String>();
+		for (int i = 0; i < instance.blockDeviceMappings().size(); i++) {
+			volumeIds.add(instance.blockDeviceMappings().get(i).ebs().volumeId());
+		}
+		return volumeIds;
+	}
+
+	protected Instance getInstanceReservation(DescribeInstancesResponse response) throws InstanceNotFoundException {
+		if (!response.reservations().isEmpty()) {
+			Reservation reservation = response.reservations().get(FIRST_POSITION);
+			if (!reservation.instances().isEmpty()) {
+				return reservation.instances().get(FIRST_POSITION);
+			}
+		}
+		throw new InstanceNotFoundException();
+	}
+	
+	protected List<InstanceNetworkInterfaceSpecification> loadNetworkInterfaces(ComputeOrder computeOrder) {
```

Comment:
this method and the next ones, that manipulate networks, will be moved to the network plugin?

Reply:
maybe I got their goal. But still, explain to me just to confirm..

Reply:
No, these methods are specific to this class.